src/app/static/src/tests/components/adr/selectDataset.test.ts

```diff
@@ -528,6 +544,37 @@ describe("select dataset", () => {
         expect(rendered.find("#loading-dataset").exists()).toBe(false);
         expect(rendered.find(Modal).props("open")).toBe(false);
     });
+    
+    it("renders reset confirmation dialog when changing selected dataset", async () => {
```

Comment:
I wanted to add a test to check that the modal was rendered once the user clicked "Save version and keep editing" in the confirmation dialog but this required me to set up so much state for the reset confirmation component that it essentially became an integration test. I don't see any integration tests for reset confirmation or selected dataset, unless I'm missing something. Let me know if you think if I should add something along these lines in.